File: 250125/바이러스 검사/virus-detector.java
import java.io.*;
import java.util.*;

// Cap: capacity
public class Main {
    public static void main(String[] args) throws IOException {
        // input
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int[] customers = new int[n];
        for (int i = 0; i < n; i++) {
            customers[i] = Integer.parseInt(st.nextToken());
        }

        st = new StringTokenizer(br.readLine());
        int leaderCap = Integer.parseInt(st.nextToken());
        int memberCap = Integer.parseInt(st.nextToken());

        long totalCount = n;

        for (int i = 0; i < n; i++) {
            customers[i] -= leaderCap;
        }

        for (int customer: customers) {
            if (customer <= 0) continue;

            int tempMemberCount = customer / memberCap;
            if (customer % memberCap == 0) {
                totalCount += tempMemberCount;
                continue;
            }

            totalCount += (tempMemberCount + 1);
        }

        bw.write(String.valueOf(totalCount));
        bw.flush();
        bw.close();
    }
}

File: 250206/바이러스 검사/virus-detector.kt
import java.util.*
import java.io.*

fun main() {
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val restaurantCount = bufferedReader.readLine().toInt()
    val restaurantCustomerToken = StringTokenizer(bufferedReader.readLine())

    val restaurantCustomers = MutableList(restaurantCount) {
        restaurantCustomerToken.nextToken().toInt()
    }

    val workerToken = StringTokenizer(bufferedReader.readLine())
    val leaderCapacity = workerToken.nextToken().toInt()
    val memberCapacity = workerToken.nextToken().toInt()

    var answer: Long = restaurantCount.toLong()

    for (customer in restaurantCustomers) {
        var remainedCustomer = customer - leaderCapacity
        if (remainedCustomer <= 0) {
            remainedCustomer = 0
            continue
        }
        
        answer += (remainedCustomer / memberCapacity)
        if (remainedCustomer % memberCapacity > 0) answer++
    }

    bufferedWriter.write("$answer")

    bufferedWriter.flush()
    bufferedWriter.close()
}
